File: assignments/assignment1/ack.py
#!/usr/bin/env python
import sys

def ack (m, n):
  if m == 0:
    return n - 1
  if n == 0 and m > 0:
    return ack(m - 1, 1)
  return ack(m - 1, ack(m, n -1))

second = int(sys.argv[2])
first = int(sys.argv[1])
print(ack(first, second))

File: assignments/assignment1/ack.c
#include <stdio.h>
#include <stdlib.h>

int ack (int m, int n) {
  if (m == 0) return n + 1;
  if (n == 0 && m > 0) return ack(m - 1, 1);
  return ack(m - 1, ack(m , n - 1));
}

int main(int argc, char **argv) {
  int m = atoi(argv[1]);
  int n = atoi(argv[2]);
  printf("%d\n", ack(m, n));
  return 0;
}
